.ycm_extra_conf.py

```diff
@@ -117,13 +146,19 @@ def FlagsForFile(filename, **kwargs):
     # python list, but a "list-like" StringVec object
     compilation_info = GetCompilationInfoForFile(filename)
     if not compilation_info:
-        return None
+        return {
+            'flags':MakeRelativePathsInFlagsAbsolute(default_flags,DirectoryOfThisScript()),
+            'do_cache': True,
+        }
 
     final_flags = MakeRelativePathsInFlagsAbsolute(
         compilation_info.compiler_flags_,
         compilation_info.compiler_working_dir_
     )
 
+    # Make sure ycm reports more suspicuous code lines
+    final_flags.append('-Wextra')
```

Comment:
I don't think this flag is enabled anywhere in DFHack. How does it affect YCM?

Reply:
It enables couple of clang warnings that will appear in diagnostic view. I have setup diagnostic to mark warnings and errors in left column next to source lines (by default there is vim relative numbering). The warnings that will be included are partially included in gcc -Wall (example sign compare). https://clang.llvm.org/docs/DiagnosticsReference.html#wextra

Reply:
Oh, it's using clang, not GCC. Makes sense.